cfme/infrastructure/pxe.py

```diff
@@ -1,96 +1,212 @@
+        tree = ManageIQTree()
+
+
+class PXEMainView(BaseLoggedInPage):
+    """
+    represents whole All PXE Servers page
+    """
+    flash = FlashMessages('.//div[@id="flash_msg_div"]/div[@id="flash_text_div" or '
+                          'contains(@class, "flash_text_div")]')
+    toolbar = View.nested(PXEToolBar)
+    sidebar = View.nested(PXESideBar)
+
+    @View.nested
+    class entities(View):  # noqa
+        """ will be filled in concrete classes"""
+
+    @property
+    def is_displayed(self):
+        return self.navigation.currently_selected == ['Compute', 'Infrastructure', 'PXE']
+
+
+class PXEServerEntity(ParametrizedView, ClickableMixin):
+    """
+     represents one PXE Server table row
+    """
+    PARAMETERS = ('name', )
+    ROOT = ParametrizedLocator('.//tr[./td[contains(normalize-space(.), {name|quote})]]')
+
+    checkbox = Checkbox(locator='./td/input[@type="checkbox"]')
+    name = Text(locator=ParametrizedString('./td[contains(normalize-space(.), {name|quote})]'))
+    last_refreshed = Text(locator=ParametrizedString('./td[contains(normalize-space(.), '
+                                                     '{name|quote})]/following-sibling::td'))
+
+    def check(self):
+        self.checkbox.fill(True)
 
-pxe_properties_form = Form(
-    fields=[
-        ('name_text', Input('name')),
-        ('log_protocol', Select("//select[@id='log_protocol']")),
-        ('userid_text', Input('log_userid')),
-        ('password_text', Input('log_password')),
-        ('verify_text', Input('log_verify')),
-        ('validate_btn', "//a[@id='val']"),
-        ('access_url_text', Input('access_url')),
-        ('uri_text', Input('uri')),
-        ('pxe_dir_text', Input('pxe_directory')),
-        ('windows_dir_text', Input('windows_images_directory')),
-        ('customize_dir_text', Input('customization_directory')),
-        ('pxe_menu_text', Input('pxemenu_0')),
-    ])
+    def uncheck(self):
+        self.checkbox.fill(False)
 
-pxe_image_type_form = Form(
-    fields=[
-        ('image_type', Select("//select[@id='image_typ']"))
-    ])
+    @property
+    def is_checked(self):
+        return self.checkbox.selected
 
 
-template_details_page = Region(infoblock_type='form')  # infoblock shoudl be type 'detail' #gofigure
+class PXEServersView(PXEMainView):
+    """
+     represents whole All PXE Servers page
+    """
+    @View.nested
+    class entities(View):  # noqa
+        title = Text('//div[@id="main-content"]//h1')
+        _table = Table(locator='.//div[@id="records_div"]/table')
 
-template_properties_form = Form(
-    fields=[
-        ('name_text', Input('name')),
-        ('description_text', Input('description')),
-        ('image_type', Select('//select[@id="img_typ"]')),
-        ('script_type', Select('//select[@id="typ"]')),
-        ('script_data', ScriptBox(ta_locator="//textarea[@id='script_data']"))
-    ])
+        @property
+        def entity_type(self):
+            return PXEServerEntity
```

Comment:
If I make it class attribute it will be picked up widgetastic and generate error.